vignettes/syntax.Rmd

```diff
@@ -0,0 +1,398 @@
+---
+title: "The tidyselect syntax"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{syntax}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+options(
+  tibble.print_min = 4,
+  tibble.print_max = 4
+)
+options(
+  crayon.enabled = FALSE
+)
+```
+
+This is a technical description of the tidyselect syntax.
+
+```{r setup}
+library(tidyselect)
+library(magrittr)
+
+# FIXME: `select_pos()` should be exported as `eval_select()`
+select_pos <- function(data, ...) {
+  tidyselect:::select_pos(data, !!rlang::expr(c(...)))
+}
+
+# For better printing
+mtcars <- tibble::as_tibble(mtcars)
+iris <- tibble::as_tibble(iris)
+```
+
+
+## Sets of variables
+
+The tidyselect syntax is all about __sets__ of variables, internally
+represented by integer vectors of __positions__. `c(1L, 2L, 1L)`
+represents the set of the first and second variables.
+
+If a vector of positions contains duplicates, they are normally
+treated as the same element since they represent sets. An exception to
+this occurs with named elements whose names differ. If the names don't
+match, they are treated as different elements in order to allow
+renaming a variable to multiple names (see section on Renaming
+variables).
+
+The syntax of tidyselect is generally designed for set combination.
+For instance, `c(foo(), bar())` represents the union of the variables
+in `foo()` and those in `bar()`.
+
+
+### Bare names
+
+Within _data-expressions_ (see Evaluation section), bare names
+represent their own positions, i.e. a set of size 1. The following
+expressions are equivalent:
+
+```{r}
+mtcars %>% select_pos(mpg:hp, -cyl, vs)
+
+mtcars %>% select_pos(1:4, -2, 8)
+```
+
```

Comment:
Maybe it's worth a mention of the other arithmetic operations somewhere? i.e. `mtcars %>% select_pos(mpg * wt)` fails because ...